content/develop/start/truffle.md

```diff
@@ -1,5 +1,156 @@
 ---
-title: "Aurora: Getting Started with Truffle"
+title: "Aurora: Deploying a Contract Using Truffle"
 ---
 
-# Getting Started with Truffle
+# Deploying a Contract Using Truffle
+
+## Introduction
+This example is originally forked from [OpenZeppelin examples](https://docs.openzeppelin.com/contracts/4.x/erc721). However it implements a simple COVID-19 vaccine token 💊💊. This tutorial assumes that you are familiar with non-fungible tokens (NFT) concept. For more details about the non-fungible token standard (ERC721), please refer to [OpenZeppelin ERC721 implementation](https://github.com/OpenZeppelin/openzeppelin-contracts/tree/1b37c21da58f6379cfe09c0140cf56d67b19a0bc/contracts/token/ERC721).
+
+## COVID Vaccine NFT
+
+![](../../_img/Truffle-nft-aurora-example.png)
+
+1. The `minter` address (which is managed by the vaccination program manager) can distribute (mint) the vaccine tickets (NFT tokens 💊💊💊) to the people who are part of the vaccination program. 
+2. When a participant receives the token 💊, he/she can get access to the vaccine by spending the NFT token. 
+3. This means whether burning the NFT token or sending it back to the minter address.
+4. Now the `minter` can redistribute that ticket 🎫 to new participant in the line.
+5. And now that participant have access to the same ticket that have been spended by participant-1.
+
+
+## Installing Prerequisites
+This tutorial assumes that you have `Node 12+` and `yarn`. Please refer to the [Yarn installation page](https://classic.yarnpkg.com/en/docs/install) if you don't have `yarn` installed locally.
+
+To install the prerequisites packages, clone the examples code then run `yarn` in order to install the prerequisites:
+```
+git clone https://github.com/aurora-is-near/aurora-examples.git
+cd aurora-examples/truffle/erc721-example/
+yarn 
+```
+
+## Connecting Truffle to Aurora
+
+Export your `MNEMONIC` as follows:
+```
+export MNEMONIC='YOUR MNEMONICS HERE'
+```
+The `truffle-config.js` will pickup your `MNEMONIC` and extract the address that will be used for sending, and signing transactions on Aurora network.
+
+Now in `truffle-config.js`, you have to change the `from` address as shown below in the aurora network section.
+
+```
+...
+aurora: {
+      provider: () => setupWallet('https://testnet.aurora.dev'),
+      network_id: 0x4e454153,
+      gas: 10000000,
+      from: '0x6A33382de9f73B846878a57500d055B981229ac4' // CHANGE THIS ADDRESS
+    },
+```
+
+## Deploying The Contract
+
+To deploy the `CovidVaccineToken` contract, you can run the yarn command as follows:
+
+```
+yarn deploy:aurora
```

Comment:
I got an error at this point

```
> erc20-example@1.0.0 compile /mnt/birchmd/tmp/aurora-examples/truffle/erc721-example
> truffle compile

sh: 1: truffle: not found
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! erc20-example@1.0.0 compile: `truffle compile`
npm ERR! spawn ENOENT
npm ERR!
npm ERR! Failed at the erc20-example@1.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/birchmd/.npm/_logs/2021-05-27T13_58_34_258Z-debug.log
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Seems like there is some missing step where `truffle` was supposed to be installed. I thought calling `yarn` above was supposed to install all the dependencies?

Reply:
Will update the doc